packages/x-charts/src/ChartsReferenceLine/ChartsXReferenceLine.tsx

```diff
@@ -105,8 +110,22 @@ function ChartsXReferenceLine(props: ChartsXReferenceLineProps) {
 
   const classes = getXReferenceLineClasses(inClasses);
 
-  const spacingX = typeof spacing === 'object' ? (spacing.x ?? 0) : spacing;
-  const spacingY = typeof spacing === 'object' ? (spacing.y ?? 0) : spacing;
+  const defaultSpacingOtherAxis =
+    labelAlign === 'middle' || labelAlign === undefined
+      ? DEFAULT_SPACING_MIDDLE_OTHER_AXIS
+      : DEFAULT_SPACING;
+
+  const spacing =
+    typeof spacingProp === 'object'
+      ? {
+          x: DEFAULT_SPACING,
+          y: defaultSpacingOtherAxis,
+          ...spacingProp,
+        }
+      : { x: spacingProp ?? DEFAULT_SPACING, y: defaultSpacingOtherAxis };
+
+  const spacingX = spacing.x;
+  const spacingY = spacing.y;
```

Comment:
Isn't this equivalent? I find it a bit easier to read:

```suggestion
  const spacingX = typeof spacingProp === 'object' ? (spacingProp.x ?? DEFAULT_SPACING) : spacingProp;
  const spacingY = typeof spacingProp === 'object' ? (spacingProp.y ?? defaultSpacingOtherAxis) : spacingProp;
```

Reply:
or possibly

```tsx
  const spacingX =
    (typeof spacingProp === 'object' ? spacingProp.x : spacingProp) ?? DEFAULT_SPACING;
  const spacingY =
    (typeof spacingProp === 'object' ? spacingProp.y : spacingProp) ?? defaultSpacingOtherAxis;
```

Reply:
Yeah, this last one looks better IMO